packages/theme/src/cli/utilities/repl/evaluater.ts

```diff
@@ -0,0 +1,199 @@
+
+  const json = `{ "type": "context", "value": "{% ${config.snippet.replace(/"/g, '\\"')} %}" }`
+  const {status, text} = await makeRequest({...config, snippet: json})
+
+  if (successfulRequest(status, text)) {
+    config.replSession.push(JSON.parse(json))
+  }
+}
+
+async function evalAssignmentContext(config: EvaluationConfig) {
+  outputDebug(`Evaluating assignment context - ${config.snippet}`)
+
+  if (isSmartAssignment(config.snippet)) {
+    config.snippet = `assign ${config.snippet}`
+    outputInfo(outputContent`${outputToken.gray(`> ${config.snippet}`)}`)
+    return evalContext(config)
+  }
+}
+
+async function evalSyntaxError(config: EvaluationConfig) {
+  outputDebug(`Evaluating syntax error - ${config.snippet}`)
+
+  let body = ''
+  if (!isStandardAssignment(config.snippet)) {
+    const {text} = await makeRequest({...config, snippet: `{{ ${config.snippet} }}`})
+    body = text
+  }
+
+  if (!hasLiquidError(body)) {
+    const {text} = await makeRequest({...config, snippet: `{% ${config.snippet} %}`})
+    body = text
+  }
+
+  if (hasLiquidError(body)) {
+    const error = body.replace(/ \(snippets\/eval line \d+\)/, '')
+    printSyntaxError(config.snippet, error)
+  }
+}
+
+function printSyntaxError(snippet: string, error: string) {
+  if (error.includes('Unknown tag')) {
+    outputInfo(outputContent`${outputToken.errorText(`Unknown object, property, tag, or filter: '${snippet}'`)}`)
+    return
+  }
+
+  const match = stripHTMLContent(error)
+  if (match && match[1]) {
+    outputInfo(outputContent`${outputToken.errorText(match[1])}`)
+  }
+}
+
+async function makeRequest(config: EvaluationConfig): Promise<{text: string; status: number}> {
+  const requestBody = buildRequestBody(config)
+  const response = await sendRenderRequest(config, requestBody)
+  return {text: await response.text(), status: response.status}
+}
+
+function buildRequestBody(config: EvaluationConfig): string {
+  const items = [...config.replSession.map((item) => JSON.stringify(item)), config.snippet]
+  return `[${items.join(',').replace(/\\"/g, '"')}]`
+}
+
+async function sendRenderRequest(config: EvaluationConfig, requestBody: string) {
+  const response = await render(config.themeSession, {
+    path: config.url,
+    query: [],
+    themeId: config.themeId,
+    cookies: '',
+    sectionId: 'announcement-bar',
+    headers: {},
+    replaceTemplates: {
+      'sections/announcement-bar.liquid': `\n{% render 'eval' %}`,
+      'snippets/eval.liquid': requestBody,
+    },
+  })
+
+  if (isExpiredSession(response.status)) {
+    expiredSessionError()
+  }
+
+  if (isTooManyRequests(response.status)) {
+    tooManyRequestsError()
+  }
+
+  const serverTiming = response.headers.get('server-timing')
+  if (isResourceNotFound(serverTiming || '')) {
+    notFoundError()
+  }
+
+  return response
+}
+
+function parseDisplayResult(result: string): string | number | undefined {
+  const match = stripHTMLContent(result)
+  if (!match || !match[1]) return
```

Comment:
(optional) We may avoid this regex if we break the line during the rendering like we're doing [here](https://github.com/Shopify/cli/blob/c0762fca1d6612469b3b2083992c975caf48e37f/packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/repl/api.rb#L54)

Personally, I believe that adding the extra `\n` is a bit nicer/deterministic because we'd no longer depend on what's being renderer inside the section, but I cannot think on any scenario this logic breaks because the `<` character is escaped, so feel free to keep it as is if you prefer 🙏